specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2020-02-02-preview/components_API.json

```diff
@@ -0,0 +1,628 @@
+            "Bluefield"
+          ],
+          "x-ms-enum": {
+            "name": "FlowType",
+            "modelAsString": true
+          }
+        },
+        "Request_Source": {
+          "type": "string",
+          "description": "Describes what tool created this Application Insights component. Customers using this API should set this to the default 'rest'.",
+          "default": "rest",
+          "enum": [
+            "rest"
+          ],
+          "x-ms-enum": {
+            "name": "RequestSource",
+            "modelAsString": true
+          }
+        },
+        "InstrumentationKey": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Application Insights Instrumentation key. A read-only value that applications can use to identify the destination for all telemetry sent to Azure Application Insights. This value will be supplied upon construction of each new Application Insights component."
+        },
+        "CreationDate": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Creation Date for the Application Insights component, in ISO 8601 format.",
+          "format": "date-time"
+        },
+        "TenantId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure Tenant Id."
+        },
+        "HockeyAppId": {
+          "type": "string",
+          "description": "The unique application ID created when a new application is added to HockeyApp, used for communications with HockeyApp.",
+          "externalDocs": {
+            "url": "https://aka.ms/ewjuel"
+          }
+        },
+        "HockeyAppToken": {
+          "type": "string",
+          "description": "Token used to authenticate communications with between Application Insights and HockeyApp.",
+          "readOnly": true,
+          "externalDocs": {
+            "url": "https://aka.ms/sls76s"
+          }
+        },
+        "provisioningState": {
+          "type": "string",
+          "description": "Current state of this component: whether or not is has been provisioned within the resource group it is defined. Users cannot change this value but are able to read from it. Values will include Succeeded, Deploying, Canceled, and Failed.",
+          "readOnly": true
+        },
+        "SamplingPercentage": {
+          "type": "number",
+          "format": "double",
+          "description": "Percentage of the data produced by the application being monitored that is being sampled for Application Insights telemetry.",
+          "externalDocs": {
+            "url": "https://aka.ms/sl0yqd"
+          }
+        },
+        "ConnectionString": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Application Insights component connection string."
+        },
+        "DisableIpMasking": {
+          "type": "boolean",
+          "description": "Disable IP masking."
+        },
+        "WorkspaceResourceId": {
```

Comment:
"DisableIpMasking" and "RetentionInDays": disappear compared to stable/2015-05-01. Are they deprecated?

Reply:
looks like this is not required. Is this property required?

Reply:
hey,
they are indeed required.
i added them again.
Thanks.

Reply:
Is `WorkspaceResourceId` required? During our offline meeting, it should be required.

Reply:
you are right. change done and commited.